gq/gq.go

```diff
@@ -51,15 +52,19 @@ type signerVerifier struct {
 // the GQ1 private numbers.
 //
 // The securityParameter parameter is the level of desired security in bits. 256 is recommended.
-func NewSignerVerifier(publicKey *rsa.PublicKey, securityParameter int) SignerVerifier {
-	n, v, nBytes, vBytes := parsePublicKey(publicKey)
+func NewSignerVerifier(publicKey *rsa.PublicKey, securityParameter int) (SignerVerifier, error) {
+	n, v, nBytes, vBytes, err := parsePublicKey(publicKey)
 	t := securityParameter / (vBytes * 8)
 
-	return &signerVerifier{n, v, nBytes, vBytes, t}
+	return &signerVerifier{n, v, nBytes, vBytes, t}, err
 }
 
-func parsePublicKey(publicKey *rsa.PublicKey) (n *big.Int, v *big.Int, nBytes int, vBytes int) {
-	n, v = publicKey.N, big.NewInt(int64(publicKey.E))
+func parsePublicKey(publicKey *rsa.PublicKey) (n *bigmod.Modulus, v *big.Int, nBytes int, vBytes int, err error) {
+	n, err = bigmod.NewModulusFromBig(publicKey.N)
+	if err != nil {
+		return
+	}
+	v = big.NewInt(int64(publicKey.E))
```

Comment:
This comment isn't requesting any changes, but I just want to note that golang enforces that publicKey.E must be fit into a 32bit int. 

This is more a question for @jonnystoten since he originally wrote this codem why the cast to int64 here? Do we need `big.NewInt(int64(publicKey.E))` or could we get away with `big.NewInt(publicKey.E)`?

```golang
// checkPub sanity checks the public key before we use it.
// We require pub.E to fit into a 32-bit integer so that we
// do not have different behavior depending on whether
// int is 32 or 64 bits. See also
// https://www.imperialviolet.org/2012/03/16/rsae.html.
func checkPub(pub *PublicKey) error {
	if pub.N == nil {
		return errPublicModulus
	}
	if pub.E < 2 {
		return errPublicExponentSmall
	}
	if pub.E > 1<<31-1 {
		return errPublicExponentLarge
	}
	return nil
}
```
https://cs.opensource.google/go/go/+/refs/tags/go1.21.4:src/crypto/rsa/rsa.go;l=95

Reply:
I believe the reason is that `big.NewInt` takes an `int64` as argument